esp-hal/src/aes/mod.rs

```diff
@@ -392,25 +431,68 @@ pub mod dma {
     }
 
     impl<'d> AesDma<'d> {
-        fn write_key(&mut self, key: impl Into<Key>) {
-            let key = key.into();
+        fn write_key(&mut self, key: &Key) {
             let key = key.as_slice();
             self.aes.write_key(key);
         }
 
+        fn write_iv(&mut self, iv: &[u8; BLOCK_SIZE]) {
+            for (word, reg) in read_words(iv).zip(self.aes.regs().iv_mem_iter()) {
+                reg.write(|w| unsafe { w.bits(word) });
+            }
+        }
+
+        fn read_iv(&self, iv: &mut [u8; BLOCK_SIZE]) {
+            iv[0..4].copy_from_slice(&self.aes.regs().iv_mem(0).read().bits().to_le_bytes());
+            iv[4..8].copy_from_slice(&self.aes.regs().iv_mem(1).read().bits().to_le_bytes());
+            iv[8..12].copy_from_slice(&self.aes.regs().iv_mem(2).read().bits().to_le_bytes());
+            iv[12..16].copy_from_slice(&self.aes.regs().iv_mem(3).read().bits().to_le_bytes());
+        }
+
+        fn start_dma_transfer(
+            &mut self,
+            number_of_blocks: usize,
+            output: &mut impl DmaRxBuffer,
+            input: &mut impl DmaTxBuffer,
+            mode: Mode,
+            cipher_mode: CipherMode,
+            key: &Key,
+        ) -> Result<(), DmaError> {
```

Comment:
Are you going to do this now for every DMA API, adding a bypass method to the actual public API that has a well defined contract?

As it creates an additional burden when changing this method, because you not only have to worry about the public API but also worry about the `AesDmaBackend` (or whatever internal API someone in future might decide to abuse this escape hatch for).

The bit that bothers me the most, is that the move API was actually designed for use cases like `AesDmaBackend`, where you need to store the transfer object in the same struct that you store the peripheral itself.

With the previous Aes DMA API, you would've NEEDed a bypass because of the lifetimes but with the move API the bypass is not needed, but you still refuse to use it 😭.